Recast/Discover/MainSearchDataSourceDelegate.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  MainSearchDataSourceDelegate.swift
+//  Recast
+//
+//  Created by Jack Thompson on 9/26/18.
+//  Copyright © 2018 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+import Kingfisher
+
+class MainSearchDataSourceDelegate: NSObject {
+
+    // MARK: - Variables
+    weak var delegate: SearchTableViewDelegate?
+    var searchResults: [PartialPodcast] = []
+
+    func fetchData(query: String) {
+        SearchEndpoint(parameters: ["term": query, "media": "podcast", "limit": -1]).run()
+            .success { response in
+                self.searchResults = response.results
+                self.delegate?.refreshController()
+            }
+            .failure { error in
+                print(error)
+        }
+    }
+
+    func resetResults() {
+        searchResults = []
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension MainSearchDataSourceDelegate: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchResults.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        //swiftlint:disable:next line_length
+        let cell = tableView.dequeueReusableCell(withIdentifier: PodcastTableViewCell.cellReuseIdentifier, for: indexPath)
```

Comment:
You should be able to just force cast this, shouldn't have to safely fall to a blank cell